ql/src/java/org/apache/hadoop/hive/ql/udf/UDFToByte.java

```diff
@@ -124,10 +125,14 @@ public ByteWritable evaluate(ShortWritable i) {
    *          The integer value to convert
    * @return Byte
    */
-  public ByteWritable evaluate(IntWritable i) {
+  public ByteWritable evaluate(IntWritable i) throws UDFArgumentException {
     if (i == null) {
       return null;
     } else {
+      int value = i.get();
+      if (value < Byte.MIN_VALUE || value > Byte.MAX_VALUE) {
+        throw new UDFArgumentException("Value out of range for Byte: " + value);
+      }
```

Comment:
I know this obeys the ANSI standard, but it is a breaking change for Hive. Hive's cast is generally known to be very generous, so I think we can't change the behavior without introducing a flag.

Reply:
Hey @okumin, thank you so much for the review!

Your suggestion makes sense. I have set the default value as true.